homeassistant/components/brother/sensor.py

```diff
@@ -25,356 +28,346 @@
-    if entity_id := entity_registry.async_get_entity_id(
-        PLATFORM, DOMAIN, old_unique_id
-    ):
-        new_unique_id = f"{coordinator.data.serial.lower()}_bw_counter"
-        _LOGGER.debug(
-            "Migrating entity %s from old unique ID '%s' to new unique ID '%s'",
-            entity_id,
-            old_unique_id,
-            new_unique_id,
-        )
-        entity_registry.async_update_entity(entity_id, new_unique_id=new_unique_id)
-
-    sensors = []
-
-    device_info = DeviceInfo(
-        configuration_url=f"http://{entry.data[CONF_HOST]}/",
-        identifiers={(DOMAIN, coordinator.data.serial)},
-        manufacturer=ATTR_MANUFACTURER,
-        model=coordinator.data.model,
-        name=coordinator.data.model,
-        sw_version=coordinator.data.firmware,
-    )
-
-    for description in SENSOR_TYPES:
-        if getattr(coordinator.data, description.key) is not None:
-            sensors.append(
-                description.entity_class(coordinator, description, device_info)
-            )
-    async_add_entities(sensors, False)
-
-
-class BrotherPrinterSensor(CoordinatorEntity, SensorEntity):
-    """Define an Brother Printer sensor."""
-
-    _attr_has_entity_name = True
-
-    def __init__(
-        self,
-        coordinator: BrotherDataUpdateCoordinator,
-        description: BrotherSensorEntityDescription,
-        device_info: DeviceInfo,
-    ) -> None:
-        """Initialize."""
-        super().__init__(coordinator)
-        self._attrs: dict[str, Any] = {}
-        self._attr_device_info = device_info
-        self._attr_unique_id = f"{coordinator.data.serial.lower()}_{description.key}"
-        self.entity_description = description
-
-    @property
-    def native_value(self) -> StateType | datetime:
-        """Return the state."""
-        return cast(
-            StateType, getattr(self.coordinator.data, self.entity_description.key)
-        )
-
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
-        """Return the state attributes."""
-        remaining_pages, drum_counter = ATTRS_MAP.get(
-            self.entity_description.key, (None, None)
-        )
-        if remaining_pages and drum_counter:
-            self._attrs[ATTR_REMAINING_PAGES] = getattr(
-                self.coordinator.data, remaining_pages
-            )
-            self._attrs[ATTR_COUNTER] = getattr(self.coordinator.data, drum_counter)
-        return self._attrs
-
-
-class BrotherPrinterUptimeSensor(BrotherPrinterSensor):
-    """Define an Brother Printer Uptime sensor."""
+@dataclass
+class BrotherSensorRequiredKeysMixin:
+    """Class for Brother entity required keys."""
 
-    @property
-    def native_value(self) -> datetime:
-        """Return the state."""
-        return cast(
-            datetime, getattr(self.coordinator.data, self.entity_description.key)
-        )
+    value: Callable[[BrotherSensors], StateType | datetime]
+    state_attrs: Callable[[BrotherSensors], dict[str, Any]]
```

Comment:
```suggestion
    extra_state_attrs: Callable[[BrotherSensors], dict[str, Any]] | None = None
```
I suggest to add the `extra_` prefix so it will be clear it is related to `_attr_extra_state_attributes`
If you add a default value you don't need to assign it every time bellow

Reply:
I think if we add a default value for `extra_state_attr` we will get "non-default argument follows default argument" error.